recipe-server/client/control/components/RecipeForm.js

```diff
@@ -54,15 +55,26 @@ export class RecipeForm extends React.Component {
 
   renderCloningMessage() {
     const isCloning = this.props.route && this.props.route.isCloning;
-    const displayedRecipe = this.props.recipe || {};
-
-    return isCloning &&
-      (<span className="cloning-message callout">
-        {'You are cloning '}
-        <Link to={`/control/recipe/${displayedRecipe.id}/`}>
-          {displayedRecipe.name} ({displayedRecipe.action})
-        </Link>.
-      </span>);
+    const displayedRecipe = this.props.recipe;
+
+    if (!isCloning || !displayedRecipe) {
+      return null;
+    }
+
+    this.cloneCache = this.cloneCache || {};
+    const cache = this.cloneCache;
+
+    if (!cache[displayedRecipe.id]) {
```

Comment:
I had noticed that I had set up the cloning message to be rendered via 
```{ this.renderCloningMessage() }```, meaning the function fires on each render. Adding the caching at least saved some cycles, but it's not a big deal if it's removed.